src/StringTools.cpp

```diff
@@ -0,0 +1,617 @@
+            while( Source[EndIndex] == CurrWhite ) EndIndex++;
+            Source.replace(CurrIndex,EndIndex - CurrIndex,1,CurrWhite);
+            CurrIndex++;
+        }
+    }
+
+    String RemoveDuplicateWhitespacesCopy(String Source)
+    {
+        StringTools::RemoveDuplicateWhitespaces(Source);
+        return Source;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Colour Channels
+
+    Real ConvertHexToColourChannel(const StringView Hex)
+    {
+        if( Hex.size() != 2 ) {
+            String ExceptionString("Hex code requires 2 characters to express a Colour channel.");
+            throw std::runtime_error(ExceptionString);
+        }
+
+        // <charconv> implementation
+        /*unsigned int Temp = 0;
+        std::from_chars_result Result = std::from_chars(Hex.data(),Hex.data() + 2,Temp,16);
+        if( Result.ec != std::errc() ) {
+            throw std::runtime_error("Error parsing Hex String. Does the String contain only valid Hex characters?");
+        }
+        Real Ret = static_cast<Real>(Temp);
+        return std::min(Ret *= HexConversionMultiplier,Real(1.0));//*/
+
+        // std::strtool implementation
+        char* EndPtr = nullptr;
+        Real Ret = static_cast<Real>( std::strtol(Hex.data(),&EndPtr,16) );
+        if( EndPtr != std::next(Hex.data(),2) ) {
+            throw std::runtime_error("Error parsing Hex String. Does the String contain only valid Hex characters?");
+        }
+        return std::min(Ret *= HexConversionMultiplier,Real(1.0));
+    }
+
+    String ConvertColourChannelToHex(const Real Channel)
+    {
+        // <charconv> implementation
+        /*String Ret(2,'0');
+        unsigned int Temp = static_cast<unsigned int>(Channel * Real(255.0));
+        std::to_chars_result Result = std::to_chars(Ret.data(),Ret.data() + 2,Temp,16);
+        if( Result.ec != std::errc() ) {
+            throw std::runtime_error("Error converting Real to Hex. Is it within the range 0.0-1.0?");
+        }
+
+        if( Result.ptr == std::next(Ret.data()) ) {
+            std::swap(Ret[0],Ret[1]);
+        }
+        StringTools::ToUpperCase(Ret.begin(),Ret.end());
+        return Ret;//*/
+
+        // StringStream implementation
+        StringStream Converter;
+        Converter.setf( std::ios::hex | std::ios::uppercase, std::ios::basefield | std::ios::uppercase );
+        Converter.width(2);
+        Converter.fill('0');
+        Converter << std::min( static_cast<unsigned int>( Channel * Real(255.0) ), 255u );
+        return Converter.str();
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Tokenizing
+
+    StringVector Split(const String& Source, const String& Delims, const Whole MaxSplits)
+    {
+        StringVector Ret;
+
+        if( Source.empty() ) {
+            return Ret;
+        }
+
+        Ret.reserve( MaxSplits ? MaxSplits + 1 : 10 );
```

Comment:
What is the use case for limiting the amount of splits? I have not seen this before.

Reply:
The default value is no maximum.  I want to say I have used the maximum somewhere in the Mezzanine, but I can't recall where.  I think this is a relic from Ogre, but unlike other Ogre relics this is doing no harm.